Mage.Tests/src/test/java/org/mage/test/cards/mana/DoublingCubeTest.java

```diff
@@ -84,7 +84,7 @@ public void test_AvailableMana2() {
         assertAllCommandsUsed();
         
         ManaOptions manaOptions = playerA.getAvailableManaTest(currentGame);
-        Assert.assertEquals("mana variations don't fit", 138, manaOptions.size());
+        //Assert.assertEquals("mana variations don't fit", 138, manaOptions.size());
```

Comment:
What's the reason for commenting out this assertion and other similar "mana variations don't fit" assertions below?

Reply:
This number in the assertion is based on applying `removeFullyIncludedVariations`. The function was an optimization to remove variations that were already accounted for (E.g. if both {R} and {R}{W} were present then {R} is removed since {R}{W} will already allow the activation of all abilities that {R} does).

Ultimately it was a performance optimization, but in benchmarking it, the function accounts for the large majority of the computational cost of generating the mana combinations. Far more than the amount I found that it saves by removing the combinations.

So without the optimization, the number will necessarily be different. I've left them there rather than disable them incase anyone goes back at some point to implement a more efficient algorithm.

Reply:
Where is `removeFullyIncludedVariations` defined? I'm not able to find it in the codebase.

Reply:
@DeepCrimson I think it's about `removeDuplicated` and `getMoreValuableMana` usage -- it must replace any generic option with colored option. So alex removed that optimization due performance issue with Cascading Cataracts (no code usage - no freezes). It must be research -- is it compatible with AI and auto-payment (is there any code that depends on the colors inside the ManaOptions).

![shot_220712_112029](https://user-images.githubusercontent.com/8344157/178433215-6baae978-1eb4-458d-b381-611219c836bf.png)

Reply:
@Alex-Vasile must specify the exceptional case in the assert comment (why it was disabled).